src/javascript/ContentEditor/utils/useReorderList.js

```diff
@@ -0,0 +1,165 @@
+    newList.splice(hoverIndex, 0, draggedItem); // Insert the dragged item at the hover position
+    return newList;
+}
+
+function makeDragItems(items) {
+    return items.map((item, index) => ({
+        item,
+        index,
+        id: uniqueId()
+    }));
+}
+
+function updateDragItems(items, reorderedItems) {
+    return items.map((item, index) => {
+        return {...reorderedItems[index], item, index};
+    });
+}
+
+/**
+ * Custom hook to manage the reordering of a list of items.
+ * It initializes the list with drag items and provides functions to handle reordering and resetting the list.
+ * @param {array} items    array of items to be reordered
+ * @returns {object}    object containing the reordered items, handleReorder function, and reset function
+ */
+export function useReorderList(items) {
+    // Note the use of serialization to detect changes in the items array
+    // this is necessary to avoid unnecessary re-renders when the items array is the same, because formik values are not stable
+    const itemsRef = useRef({items, serialized: JSON.stringify(items)});
+    const currentItemsSerialized = JSON.stringify(items);
+
+    const [reorderedItems, setReorderedItems] = useState(makeDragItems(items));
+
+    const reset = useCallback(() => {
+        setReorderedItems(makeDragItems(items));
+        // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, [currentItemsSerialized]);
+
+    const handleReorder = useCallback((dragIndex, hoverIndex) => {
+        setReorderedItems(prevItems =>
+            onListIndexReorder(prevItems, dragIndex, hoverIndex).map((item, index) => ({...item, index}))
+        );
+    }, []);
+
+    if (currentItemsSerialized !== itemsRef.current.serialized) {
+        /*
+        The significance of this check is to have latest values available immediately, without an extra render
+         */
+        let newReorderedItems;
+        if (itemsRef.current.items.length === items.length) {
+            newReorderedItems = updateDragItems(items, reorderedItems);
+        } else {
+            newReorderedItems = makeDragItems(items);
+        }
+
+        setReorderedItems(newReorderedItems);
+        itemsRef.current = {items, serialized: JSON.stringify(items)};
+        return {reorderedItems: newReorderedItems, handleReorder, reset};
+    }
+
+    return {
+        reorderedItems,
+        handleReorder,
+        reset
+    };
+}
+
+/**
+ * Custom hook to handle the drop event for reordering items.
+ * @param {object} args    object contains the following properties:
+ * @param {object} args.ref    reference to the drop target
+ * @param {number} args.index    index of the item being hovered
+ * @param {function} args.onReorder    function to call when the item is reordered during hover
+ * @param {object} useDropArgs    additional arguments for the useDrop hook
+ */
+export function useReorderDrop(args, useDropArgs) {
+    const hover = (item, monitor) => {
```

Comment:
I'm guessing the whole purpose of the custom hook for drop is to modify the hover ? Can we explain a bit what we're exactly doing here as a comment